typing/jkind.ml

```diff
@@ -305,13 +305,19 @@ module Layout = struct
            to end up less than a sort (so, no [any]), but it seems easier to keep
            this case lined up with the inverse case, which definitely cannot use
            [to_product_sort]. *)
-        match Sort.decompose_into_product s2 (List.length ts1) with
+        match
+          Sort.decompose_into_product ~level:Btype.generic_level s2
```

Comment:
I'm surprised to see `generic_level` here and below - I know there are some optimizations for types where the typechecker eagerly picks generic level and trusts it be lowered if needed, but I couldn't convince myself that would work here. Was this intentional?

If not `generic_level`, I think the right thing is probably `current_level`. Which you don't easily have access to here, so I think adding another level parameter would be fine.  I think probably we'll eventually want to just give this file access to `current_level` with a ref to resolve the circularity, and eliminate some of the level parameters, but they are fine for now.